framework/src/transfers/MultiAppPostprocessorNearestElementTransfer.C

```diff
@@ -0,0 +1,136 @@
+
+  // Check that the variable is a CONSTANT MONOMIAL.
+  auto & to_fe_type = _to_sys.variable_type(_to_var_num);
+  if (to_fe_type.order != CONSTANT || to_fe_type.family != MONOMIAL)
+    mooseError(parameters.blockLocation() + " " + parameters.blockFullpath() +
+               "\nMultiAppPostprocessorNearestElementTransfer works with CONSTANT MONOMIAL "
+               "variables only");
+}
+
+void
+MultiAppPostprocessorNearestElementTransfer::initialSetup()
+{
+  // Cache the Multiapp position ID for every element.
+  unsigned int multiapp_pos_id = 0;
+  for (auto & elem :
+       as_range(_to_mesh.active_local_elements_begin(), _to_mesh.active_local_elements_end()))
+  {
+    // Exclude the elements without dofs.
+    if (elem->n_dofs(_to_sys_num, _to_var_num) > 0)
+    {
+      Real distance = std::numeric_limits<Real>::max();
+      for (unsigned int j = 0; j < getFromMultiApp()->numGlobalApps(); ++j)
+      {
+        Real current_distance = (getFromMultiApp()->position(j) - elem->true_centroid()).norm();
+        if (current_distance < distance)
+        {
+          distance = current_distance;
+          multiapp_pos_id = j;
+        }
+      }
+      _cached_multiapp_pos_ids.push_back(multiapp_pos_id);
+    }
+  }
+}
+
+void
+MultiAppPostprocessorNearestElementTransfer::execute()
+{
+  // Check that transfer direction is FROM_MULTIAPP.
+  if (_current_direction != FROM_MULTIAPP)
+    paramError("direction", "This transfer works only with direction equal to FROM_MULTIAPP");
+
+  // Retrieve the vector of the variable values.
+  NumericVector<Real> & solution = *_to_sys.solution;
+
+  // Store the local multiapps postprocessor values.
+  const unsigned int n_subapps = getFromMultiApp()->numGlobalApps();
+  std::vector<Real> pp_values(n_subapps, std::numeric_limits<Real>::max());
+  std::vector<Real> duplicate(n_subapps, -std::numeric_limits<Real>::max());
+  for (const auto i : make_range(n_subapps))
+    if (getFromMultiApp()->hasLocalApp(i))
+    {
+      pp_values[i] = getFromMultiApp()->appPostprocessorValue(i, _postprocessor_name);
+      duplicate[i] = pp_values[i];
+    }
+
+  // Gather all the multiapps postprocessor values.
+  _communicator.min(pp_values);
+  _communicator.max(duplicate);
+  for (const auto i : make_range(n_subapps))
+    if (pp_values[i] != duplicate[i])
+      mooseError("There should be only one processor setting a subapp postprocessor but now this "
+                 "appears not true.");
+
+  // Assign the multiapps postprocessor values to the local elements.
+  unsigned int i = 0;
+  for (auto & elem :
+       as_range(_to_mesh.active_local_elements_begin(), _to_mesh.active_local_elements_end()))
+  {
+    // Exclude the elements without dofs
+    if (elem->n_dofs(_to_sys_num, _to_var_num) > 0)
+    {
+      dof_id_type dof = elem->dof_number(_to_sys_num, _to_var_num, 0);
+      auto v = solution(dof);
+      Real nv = v * (1 - _relax_factor) + pp_values[_cached_multiapp_pos_ids[i]] * _relax_factor;
```

Comment:
This is extremely hacking. I would like to see transfer only does transfer.


May separate it  into a pluggable system

Reply:
I might not understand the motivation here